packages/animations/example/lib/container_transition.dart

```diff
@@ -0,0 +1,419 @@
+                        openContainer: openContainer,
+                        subtitle: 'Secondary text',
+                      );
+                    },
+                  ),
+                ),
+              ],
+            ),
+            const Padding(
+              padding: EdgeInsets.symmetric(vertical: 8.0),
+            ),
+            Row(
+              children: <Widget>[
+                Expanded(
+                  child: _OpenContainerWrapper(
+                    closedBuilder:
+                        (BuildContext context, VoidCallback openContainer) {
+                      return _SmallerCard(
+                        openContainer: openContainer,
+                        subtitle: 'Secondary',
+                      );
+                    },
+                  ),
+                ),
+                const Padding(
+                  padding: EdgeInsets.symmetric(horizontal: 4.0),
+                ),
+                Expanded(
+                  child: _OpenContainerWrapper(
+                    closedBuilder:
+                        (BuildContext context, VoidCallback openContainer) {
+                      return _SmallerCard(
+                        openContainer: openContainer,
+                        subtitle: 'Secondary',
+                      );
+                    },
+                  ),
+                ),
+                const Padding(
+                  padding: EdgeInsets.symmetric(horizontal: 4.0),
+                ),
+                Expanded(
+                  child: _OpenContainerWrapper(
+                    closedBuilder:
+                        (BuildContext context, VoidCallback openContainer) {
+                      return _SmallerCard(
+                        openContainer: openContainer,
+                        subtitle: 'Secondary',
+                      );
+                    },
+                  ),
+                ),
+              ],
+            ),
+            ...List<Widget>.generate(10, (int index) {
+              return _OpenContainerWrapper(
+                closedBuilder:
+                    (BuildContext context, VoidCallback openContainer) {
+                  return ListTile(
+                    leading: Image.asset(
+                      'assets/avatar_logo.png',
+                      width: 40,
+                    ),
+                    onTap: openContainer,
+                    title: Text('List item ${index + 1}'),
+                    subtitle: const Text('Secondary text'),
+                  );
+                },
+              );
+            }),
+          ],
+        ),
+      ),
+      floatingActionButton: OpenContainer(
+        openBuilder: (BuildContext context, VoidCallback _) {
+          return _DetailsPage();
+        },
+        closedElevation: 6.0,
+        closedShape: const CircleBorder(),
+        closedBuilder: (BuildContext context, VoidCallback openContainer) {
+          return FloatingActionButton(
```

Comment:
Instead of throwing a real floating action button in here the current setup of this widget requires you to "recreate" a floating action button: Instead of returning a FloatingActionButton, just return the Icon properly aligned and sized and then use the settings on the OpenContainer to style it.

Reply:
What you did works, but doing it this way will give an even nicer transition.

Reply:
I think I'm going to stick with using this implementation for now -- I got started trying to recreate the button and it seemed to be more difficult than I expected it to be